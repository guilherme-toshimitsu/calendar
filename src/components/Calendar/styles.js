import {memo} from 'react';
import styled from 'styled-components';

const WeekdaysDiv = styled.div`
	overflow: hidden;
	padding: 10px;
`;

export {WeekdaysDiv};
