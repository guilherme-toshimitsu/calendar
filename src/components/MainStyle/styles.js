const styles = (theme) => ({
	root: {
		position: 'relative',
		textAlign: 'center',
		minHeight: '100vh',
		minWidth: 320,
		overflow: 'hidden',
		backgroundColor: theme.mainStyle.mainBackgroundColor,
	},
});

export default styles;
