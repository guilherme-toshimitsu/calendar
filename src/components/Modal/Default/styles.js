const styles = () => ({
  root: {
    width: 400,
    height: 400,
  },
});

export default styles;
