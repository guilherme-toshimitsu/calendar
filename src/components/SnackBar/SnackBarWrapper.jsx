import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

import SnackBar from './SnackBar';
import { SnackBarProvider } from './snackContext';

const initialValuesSnackBar = {
  vertical: 'top',
  horizontal: 'right',
  snackbarType: 'success',
  message: 'general.ok',
  autoClose: false,
  action: null,
  onSuccess: () => {},
};

const SnackBarWrapper = ({ children }) => {
  const [snackValues, setSnackValues] = useState(initialValuesSnackBar);
  const [open, setOpen] = useState(false);

  const openSnackBar = (val) => {
    const snack = Object.assign(initialValuesSnackBar, val);
    setSnackValues(snack);
    setOpen(true);
  };

  const onSuccess = () => {
    snackValues.onSuccess();
    setOpen(false);
  };

  useEffect(() => {
    if (open === true && snackValues.autoClose) {
      setTimeout(() => {
        setOpen(false);
        if (snackValues.snackbarType === 'success') {
          snackValues.onSuccess();
        }
      }, snackValues.autoClose);
    }
  }, [open, snackValues.autoClose, setOpen, snackValues]);

  return (
    <SnackBarProvider
      value={{
        snackValues: snackValues,
        openSnackBar,
      }}
    >
      <SnackBar
        open={open}
        onClose={() => setOpen(false)}
        onSuccess={() => onSuccess()}
        vertical={snackValues.vertical}
        horizontal={snackValues.horizontal}
        snackbarType={snackValues.snackbarType}
        action={snackValues.action}
        message={snackValues.message}
      />
      {children}
    </SnackBarProvider>
  );
};

SnackBarWrapper.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element),
  ]).isRequired,
};

export default SnackBarWrapper;
