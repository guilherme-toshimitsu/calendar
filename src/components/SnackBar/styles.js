import green from '@material-ui/core/colors/green';
import amber from '@material-ui/core/colors/amber';
import blue from '@material-ui/core/colors/blue';

const snackBarContentStyles = (theme) => ({
  success: {
    backgroundColor: green[600],
  },
  error: {
    backgroundColor: theme.palette.error.dark,
  },
  info: {
    backgroundColor: blue[600],
  },
  warning: {
    backgroundColor: amber[700],
  },
  icon: {
    fontSize: 20,
  },
  iconVariant: {
    opacity: 0.9,
    marginRight: theme.spacing(),
  },
  message: {
    display: 'flex',
    alignItems: 'center',
    alignText: 'left',
  },
});

const snackBarStyles = (theme) => ({
  root: {
    maxWidth: '70%',
  },
  close: {
    padding: theme.spacing() / 2,
  },
});

export { snackBarStyles, snackBarContentStyles };
