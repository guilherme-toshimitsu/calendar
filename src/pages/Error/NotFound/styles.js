const styles = () => ({
  root: {
    width: '100%',
    height: '100%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  notFoundText: {
    fontSize: 24,
    color: 'red',
  },
});

export default styles;
