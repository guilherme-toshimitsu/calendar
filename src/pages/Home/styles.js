const styles = (theme) => ({
  root: {
    maxWidth: 1360,
    width: '70%',
    margin: 'auto',
    padding: 40,
  },
  title: {
    textAlign: 'left',
    fontSize: 24,
    color: theme.mainStyle.mainThemeColor,
    fontStyle: 'bold',
  },
  text: {
    textAlign: 'left',
    fontSize: 16,
    marginTop: 16,
  },
  subContainer: {
    marginTop: 32,
    marginBottom: 32,
  },
});

export default styles;
