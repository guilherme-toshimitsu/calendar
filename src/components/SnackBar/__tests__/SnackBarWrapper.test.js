import React from 'react';
import { render, act, fireEvent } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { createMuiTheme } from '@material-ui/core';
import { useSnackBar } from '../';

import SnackBarWrapper from '../SnackBarWrapper';

const theme = {};

describe('<SnackBarWrapper />', () => {
  test('if render', () => {
    const { asFragment } = render(
        <SnackBarWrapper>
          <div />
        </SnackBarWrapper>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('change context', () => {
    const Euclides = () => {
      const { openSnackBar } = useSnackBar();

      return (
        <div
          data-testid="div1"
          onClick={() => {
            openSnackBar({
              snackbarType: 'error',
              message: 'error.default.404',
              autoClose: 3000,
            });
          }}
        />
      );
    };

    act(() => {
      const { getByTestId } = render(
          <SnackBarWrapper>
            <Euclides />
          </SnackBarWrapper>
      );

      fireEvent.click(getByTestId('div1'));
    });
  });

  test('onSuccess context', () => {
    const Euclides = () => {
      const { openSnackBar } = useSnackBar();

      return (
        <div
          data-testid="div1"
          onClick={() => {
            openSnackBar({
              snackbarType: 'success',
              message: 'success',
              autoClose: 1,
              onSuccess: () => {},
            });
          }}
        />
      );
    };

    act(() => {
      const { getByTestId } = render(
          <SnackBarWrapper>
            <Euclides />
          </SnackBarWrapper>
      );

      fireEvent.click(getByTestId('div1'));
    });
  });

  test('error with autoclose', () => {
    const Euclides = () => {
      const { openSnackBar } = useSnackBar();

      return (
        <div
          data-testid="div1"
          onClick={() => {
            openSnackBar({
              snackbarType: 'error',
              message: 'success',
              autoClose: 1,
              onSuccess: () => {},
            });
          }}
        />
      );
    };

    act(() => {
      const { getByTestId } = render(
          <SnackBarWrapper>
            <Euclides />
          </SnackBarWrapper>
      );

      fireEvent.click(getByTestId('div1'));
    });
  });

  test('onSuccess without autoclose', () => {
    const Euclides = () => {
      const { openSnackBar } = useSnackBar();

      return (
        <div
          data-testid="div1"
          onClick={() => {
            openSnackBar({
              snackbarType: 'success',
              message: 'success',
              onSuccess: () => {},
            });
          }}
        />
      );
    };

    const { getByTestId } = render(
        <SnackBarWrapper>
          <Euclides />
        </SnackBarWrapper>
    );

    fireEvent.click(getByTestId('div1'));

    act(() => {
      fireEvent.click(getByTestId('snackbar-icon-button-close'));
    });
  });

  test('onClose without autoclose', () => {
    const Euclides = () => {
      const { openSnackBar } = useSnackBar();

      return (
        <div
          data-testid="div1"
          onClick={() => {
            openSnackBar({
              snackbarType: 'error',
              message: 'success',
              onSuccess: () => {},
            });
          }}
        />
      );
    };

    const { getByTestId } = render(
        <SnackBarWrapper>
          <Euclides />
        </SnackBarWrapper>
    );

    fireEvent.click(getByTestId('div1'));

    act(() => {
      fireEvent.click(getByTestId('snackbar-icon-button-close'));
    });
  });

  test('onSuccess without autoclose default function', () => {
    const Euclides = () => {
      const { openSnackBar } = useSnackBar();

      return (
        <div
          data-testid="div1"
          onClick={() => {
            openSnackBar({
              snackbarType: 'success',
              message: 'success',
            });
          }}
        />
      );
    };

    const { getByTestId } = render(
      <ThemeProvider theme={createMuiTheme(theme)}>
        <SnackBarWrapper>
          <Euclides />
        </SnackBarWrapper>
      </ThemeProvider>
    );

    fireEvent.click(getByTestId('div1'));

    act(() => {
      fireEvent.click(getByTestId('snackbar-icon-button-close'));
    });
  });
});
