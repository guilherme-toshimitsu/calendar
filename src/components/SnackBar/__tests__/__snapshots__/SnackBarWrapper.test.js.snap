// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SnackBarWrapper /> if render 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
